collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -451,8 +451,8 @@ public void recordBean(
           Object beanValue) {
 
         String beanName = domain + angleBrackets(beanProperties.toString()) + angleBrackets(attrKeys.toString());
-        // attrDescription tends not to be useful, so give the fully qualified name too.
-        String help = attrDescription + " (" + beanName + attrName + ")";
+        // attrDescription tends not to be useful, so include domain and attrName.
+        String help = attrDescription + " (domain=" + domain + ",attrName=" attrName + ")";
```

Comment:
What do you think of this?
```java
        String help = domain + ":name=" + beanProperties.get("name") + ",type=" + beanProperties.get("type") + ",attribute=" + attrName;
        if (attrDescription != null && !attrDescription.equals(attrName)) {
          help = attrDescription + " " + help;
        }
```
Having the domain, type, name, and attribute would allow users to uniquely identify where the data is coming from, while it still does not contain the `beanName` which was causing the cardinality explosion.

Example output:
```
# HELP java_lang_Metaspace_CollectionUsageThresholdSupported java.lang:name=Metaspace,type=MemoryPool,attribute=CollectionUsageThresholdSupported
# TYPE java_lang_Metaspace_CollectionUsageThresholdSupported untyped
java_lang_Metaspace_CollectionUsageThresholdSupported{type="MemoryPool",} 0.0
```

Reply:
I like it. Let me know if those comments make sense.

Reply:
:+1: thanks a lot!